wps_tools/output_handling.py

```diff
@@ -0,0 +1,181 @@
+import json, requests, math
+
+from netCDF4 import Dataset
+from tempfile import NamedTemporaryFile
+from bs4 import BeautifulSoup
+from urllib.request import urlopen, urlretrieve
+from rpy2 import robjects
+
+from wps_tools.file_handling import copy_http_content
+from wps_tools.R import load_rdata_to_python, get_package
+
+
+def nc_to_dataset(url):
+    """
+    Access content of a netcdf file from an http url as a Dataset object
+    using the netCDF4 library.
+
+    Parameters:
+        url (str): http url path to a netCDF file
+
+    Returns:
+        Dataset: Dataset object containing input netCDF file content
+    """
+
+    with NamedTemporaryFile(
+        suffix=".nc", prefix="tmp_copy", dir="/tmp", delete=True
+    ) as tmp_file:
+
+        return Dataset(copy_http_content(url, tmp_file))
```

Comment:
This question is shared across a number of the methods here, at what point are the temp files deleted?

Reply:
I'm pretty sure that the file it deleted at the end of the `with` block. 

```quote
"If delete is true (the default), the file is deleted as soon as it is 
closed. The returned object is always a file-like object whose
file attribute is the underlying true file object. This file-like object
can be used in a with statement, just like a normal file." 
```
from the [tempfile docs](https://docs.python.org/3/library/tempfile.html)